earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -196,44 +300,45 @@ WriteDocumentToString(DOMDocument *doc, std::string &buf) throw()
       if (writer->getDomConfig()->canSetParameter(XMLUni::fgDOMWRTDiscardDefaultContent, true))
-        writer->getDomConfig()->setParameter(XMLUni::fgDOMWRTDiscardDefaultContent, true);
+          writer->getDomConfig()->setParameter(XMLUni::fgDOMWRTDiscardDefaultContent, true);
       if (writer->getDomConfig()->canSetParameter(XMLUni::fgDOMWRTFormatPrettyPrint, true))
-        writer->getDomConfig()->setParameter(XMLUni::fgDOMWRTFormatPrettyPrint, true);
+          writer->getDomConfig()->setParameter(XMLUni::fgDOMWRTFormatPrettyPrint, true);
 
       MemBufFormatTarget formatTarget;
       DOMLSOutput* lsOutput = impl->createLSOutput();
       lsOutput->setByteStream(&formatTarget);
       if (writer->write(doc, lsOutput)) {
-        buf.append((const char *)formatTarget.getRawBuffer(),
-                   formatTarget.getLen());
+          buf.append((const char *)formatTarget.getRawBuffer(),
+          formatTarget.getLen());
         success = true;
       } else {
         notify(NFY_WARN, "Unable to write XML to string: Xerces didn't tell me why not.");
       }
-      lsOutput->release();
-    } catch (const XMLException& toCatch) {
-      notify(NFY_WARN, "Unable to write XML: %s",
-             XMLString::transcode(toCatch.getMessage()));
-    } catch (const DOMException& toCatch) {
-      notify(NFY_WARN, "Unable to write XML: %s",
-             XMLString::transcode(toCatch.msg));
-    } catch (...) {
-      notify(NFY_WARN, "Unable to write XML: Unknown exception");
-    }
-    writer->release();
-  } catch (const XMLException& toCatch) {
-    notify(NFY_WARN, "Unable to create DOM Writer: %s",
-           XMLString::transcode(toCatch.getMessage()));
-  } catch (const DOMException& toCatch) {
-    notify(NFY_WARN, "Unable to create DOM writer: %s",
-           XMLString::transcode(toCatch.msg));
-  } catch (...) {
-    notify(NFY_WARN, "Unable to create DOM writer: Unknown exception");
-  }
-
+        lsOutput->release();
+      } catch (const XMLException& toCatch) {
+        notify(NFY_WARN, "Unable to write XML: %s",
+               XMLString::transcode(toCatch.getMessage()));
+      } catch (const DOMException& toCatch) {
+        notify(NFY_WARN, "Unable to write XML: %s",
+               XMLString::transcode(toCatch.msg));
+      } catch (...) {
+        notify(NFY_WARN, "Unable to write XML: Unknown exception");
+      }
+      writer->release();
+      } catch (const XMLException& toCatch) {
+        notify(NFY_WARN, "Unable to create DOM Writer: %s",
+               XMLString::transcode(toCatch.getMessage()));
+      } catch (const DOMException& toCatch) {
+        notify(NFY_WARN, "Unable to create DOM writer: %s",
+               XMLString::transcode(toCatch.msg));
+      } catch (...) {
+        notify(NFY_WARN, "Unable to create DOM writer: Unknown exception");
+      }
+  //}
   return success;
 }
 
+// eventually lock parser from creation to destruction
```

Comment:
I don't understand this comment. It sounds like a bad idea to hold a lock from the time the parser is created to the time it is destroyed.

Reply:
I guess this gets down to implementation. It doesn't really pertain to this PR and can be discussed with the next PR